specs/scheduler.go

```diff
@@ -0,0 +1,62 @@
+package specs
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+)
+
+type Scheduler int
+
+const (
+	SchedulerDFS Scheduler = iota
+	SchedulerRoundRobin
+)
+
+var AllSchedulers = Schedulers{SchedulerDFS, SchedulerRoundRobin}
+var AllSchedulerNames = []string{"dfs", "round-robin"}
+
+type Schedulers []Scheduler
+
+func (s Schedulers) String() string {
+	var buffer bytes.Buffer
+	for i, scheduler := range s {
+		if i > 0 {
+			buffer.WriteString(", ")
+		}
+		buffer.WriteString(scheduler.String())
+	}
+	return buffer.String()
+}
+
+func (s Scheduler) String() string {
+	return AllSchedulerNames[s]
```

Comment:
To be more confident, here's a little trick:
```go
var AllSchedulerNames = [...]string{
	SchedulerDFS:        "dfs",
	SchedulerRoundRobin: "round-robin",
}
```

https://go.dev/play/p/IDDk0f9sBpw

Reply:
Interesting, I hadn't seen that syntax before!